README.md

```diff
@@ -341,8 +341,17 @@ See [docs/gomod.md](docs/gomod.md) for more details.
   your `go.mod` file yourself (i.e. you only specify the language release) Go will try to correct
   it automatically inside the file. Last but not least, Go 1.21 also introduced a new keyword
   [`toolchain`](https://go.dev/ref/mod#go-mod-file-toolchain) to the `go.mod` file. What this all
-  means in practice for end users is that you may not be able to process your `go.mod` file with an older version
-  of Go (and hence older cachi2) as you could in the past for various reasons.
+  means in practice for end users is that you may not be able to process your `go.mod` file with an
+  older version of Go (and hence older cachi2) as you could in the past for various reasons.
+  Many projects bump their required Go toolchain's micro release as soon as it becomes available
+  upstream (i.e. not waiting for distributions to bundle them properly). This caused problems for
+  *cachi2-v0.5.0* because the container image's version simply may not have been high enough to
+  process a given project's `go.mod` file. Therefore, *cachi2-v0.6.0* implemented a mechanism to
+  always rely on the origin 0th release of a toolchain (e.g. 1.21.0) and use the `GOTOOLCHAIN=auto`
+  setting to instruct Go to fetch any toolchain as specified by the `go.mod` file automatically,
+  hence allowing us to keep up with frequent micro version bumps. **Note that such a language
+  version would still need to be officially marked as supported by cachi2, i.e. we'd not allow Go
+  to fetch e.g. a 1.22 toolchain if the maximum supported Go version by cachi2 were 1.21!**
```

Comment:
> which means [this](https://github.com/containerbuildsystem/cachi2/blob/278c0b241ad2aae69bd18be57409de80e7a7ea03/README.md#user-content-fn-go-compat-8d98370aad2c1c2101383579142caa1f) footnote is now incorrect

Technically, it was incorrect all along because any new keywords from releases newer than cachi2's base image would have still failed, I believe that's why it uses the 'there's a good chance' wording. I think it's appropriate to re-word this footnote to accurately reflect the reality of 1.21+.

> If you feel like this is important to be documented, why don't we move this to the docs/gomod.md specific docs?

I believe it is because it has a noticeable user impact, or at least it may have had based on the failures they'd get from requiring latest toolchain micro releases. Okay, I'll try to move it to the gomod docs and add a reference to that in the main README file.

Reply:
Changed, let me know if you agree with the latest docs changes.